src/Chunker.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * LSST Data Management System
+ * Copyright 2014-2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_SG_CHUNKER_H_
+#define LSST_SG_CHUNKER_H_
+
+/// \file
+/// \author Serge Monkewitz
+/// \brief This file declares a class for partitioning the sky into chunks
+///        and sub-chunks.
+
+#include <stdint.h>
+#include <vector>
+
+#include "Angle.h"
+#include "Box.h"
+
+
+namespace lsst {
+namespace sg {
+
+/// `SubChunks` represents a set of sub-chunks of a particular chunk.
+///
+/// TODO(smm): implement a more memory efficient representation than this.
+struct SubChunks {
+    int32_t chunkId;
+    std::vector<int32_t> subChunkIds;
+
+    SubChunks() : chunkId(-1) {}
+
+    void swap(SubChunks & sc) {
+        std::swap(chunkId, sc.chunkId);
+        subChunkIds.swap(sc.subChunkIds);
+    }
+};
+
+
+/// `Chunker` subdivides the sphere into longitude-latitude boxes.
+///
+/// The celestial sphere is divided into latitude angle "stripes" of fixed
+/// height H. For each stripe, a width W is computed such that any two points
+/// in the stripe with longitude angles separated by at least W have angular
+/// separation of at least H. The stripe is then broken into an integral number
+/// of chunks of width at least W. The same procedure is used to obtain finer
+/// subchunks - each stripe is broken into a configureable number of
+/// equal-height "substripes", and each substripe is broken into equal-width
+/// subchunks.
+class Chunker {
+public:
+    Chunker(Angle overlap,
+            int32_t numStripes,
+            int32_t numSubStripesPerStripe);
+    ~Chunker();
+
+    Angle getOverlap() const {
+        return _overlap;
+    }
+    int32_t getNumStripes() const {
+        return _numStripes;
+    }
+    int32_t getNumSubStripesPerStripe() const {
+        return _numSubStripesPerStripe;
+    }
```

Comment:
The above methods could use some documentation (especially `getOverlap()`, as I think I could guess what the others do).